src/org/skypro/skyshop/product/SearchEngine.java

```diff
@@ -1,40 +1,34 @@
 package org.skypro.skyshop.product;
 
 import java.util.Arrays;
+import java.util.LinkedList;
 
 public class SearchEngine {
 
-    private final Searchable[] searchables;
+    LinkedList<Searchable> searchables;
     private int count = 0;
 
     public SearchEngine(int size) {
-        searchables = new Searchable[size];
+        searchables = new LinkedList<>();
     }
 
     public void add(Searchable item) {
-        if (count < searchables.length) {
-            searchables[count++] = item;
-        }
+        searchables.add(item);
+        count++;
     }
 
-    public Searchable[] search(String term) {
-        Searchable[] arr = new Searchable[5];
-        int resultCount = 0;
-
+    public LinkedList<Searchable> search(String term) {
```

Comment:
```suggestion
    public List<Searchable> search(String term) {
```
помним про полиморфизм